projects/storefrontlib/cms-components/myaccount/new-notification-preference/new-notification-preference.component.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import {
+  NotificationPreference,
+  UserNotificationPreferenceService,
+} from '@spartacus/core';
+import { combineLatest, Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-new-notification-preference',
+  templateUrl: './new-notification-preference.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NewNotificationPreferenceComponent implements OnInit {
+  preferences$: Observable<NotificationPreference[]>;
+  isLoading$: Observable<boolean>;
+
+  protected preferences: NotificationPreference[] = [];
+
+  constructor(
+    private notificationPreferenceService: UserNotificationPreferenceService
```

Comment:
Please use `protected` instead of `private` to make this easier for users to extend.

Reply:
modified as suggestion.Only after CDC/CDP changing cms config and introduce `newNotificationPreference' in the future, it will be displayed on SPA UI. That's why there is no e2e test. But we already have a ticket to track it.
https://jira.tools.sap/browse/CXSPA-4475